### PR TITLE
fix the readthedocs build, enable readthedocs PR builds and move readthedocs config into .readthedocs.yml

### PR DESCRIPTION
**Pull request by graingert**
## Contributor Checklist:

* [x] The associated ticket in Trac is here: https://twistedmatrix.com/trac/ticket/9980
* [ ] I ran `tox -e black-reformat` to format my patch to meet the [Twisted Coding Standard](https://twistedmatrix.com/documents/current/core/development/policy/coding-standard.html)
* [ ] I ran additional checks listed at [Getting Your Patch Accepted](https://twistedmatrix.com/trac/wiki/TwistedDevelopment#GettingYourPatchAccepted)
* [x] I have created a newsfragment in src/twisted/newsfragments/ (see: [News files](https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles))
* [x] I have updated the automated tests.
* [x] I have submitted the associated Trac ticket for review by adding the word `review` to the keywords field in Trac, and putting a link to this PR in the comment; it shows up in https://twisted.reviews/ now.

### CODE REVIEW

**Review by psi29a — APPROVED** [review]

**Comment by graingert** [comment]
@psi29a this PR is waiting on github configuration changes, did you mean to review https://github.com/twisted/ldaptor/pull/185/files instead?

File: docs/rtd-requirements.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+setuptools~=50.3
+alabaster~=0.7.12
+commonmark~=0.9.1
+docutils~=0.16.0
+mock~=4.0
+pillow~=7.2
+readthedocs-sphinx-ext~=2.1
+recommonmark~=0.6.0
+sphinx~=3.2
+sphinx-rtd-theme~=0.5.0
```
Comment (graingert) [inline docs/rtd-requirements.txt]: I like semver ranges, but I know others prefer unlimited version ranges.

also should these be in a `[docs]` extra? I don't want to put setuptools in an extra though

File: .readthedocs.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  - pdf
+  - epub
+
+python:
+  version: 3.8
+  system_packages: False
+  install:
+    - requirements: docs/rtd-requirements.txt
+    - method: pip
+      path: .
```
Comment (adiroiban) [inline .readthedocs.yml:15]: I think that we should just use `.[dev]` to install the dependencies.

File: src/twisted/newsfragments/9980.misc [hunk 3]
```diff
@@ -0,0 +1 @@
+fix the readthedocs build, enable readthedocs PR builds and move readthedocs config into .readthedocs.yml
```
Comment (adiroiban) [inline src/twisted/newsfragments/9980.misc]: I think that the release note should not read like a commit message, but rather as a brief / once sentence / non exhaustive summary / description of the changes in this PR.

So maybe just `Fix Read The Docs build and enabled PR builds.`

File: .readthedocs.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+formats:
+  - pdf
+  - epub
+
+python:
+  version: 3.8
+  system_packages: False
+  install:
+    - requirements: docs/rtd-requirements.txt
```
Comment (adiroiban) [inline .readthedocs.yml]: I would prefer to have all the dependencies defined in a single file.

Right now, all the deps are defined in src/twisted/python/_setup.py ...

So if you think that we need a separate list of `rtd` deps, it should be defined there.

But I am +1 on just using the current `dev` list of dependencies

**Review by adiroiban — CHANGES_REQUESTED** [review]
Thanks. Happy to see the RTD build.

Please see my inline comments.

File: src/twisted/newsfragments/9980.misc (hunk 3 above)
Comment (graingert) [reply src/twisted/newsfragments/9980.misc]:
```suggestion
Fix Read The Docs build and enabled PR builds.
```

File: src/twisted/python/test/test_setup.py [hunk 5]
```diff
@@ -127,12 +127,26 @@ def test_extrasRequiresDevDeps(self):
         L{_EXTRAS_REQUIRE}'s C{dev} extra contains setuptools requirements for
         the tools required for Twisted development.
         """
-        deps = _EXTRAS_REQUIRE["dev"]
-        self.assertIn("pyflakes >= 1.0.0", deps)
-        self.assertIn("twisted-dev-tools >= 0.0.2", deps)
-        self.assertIn("python-subunit", deps)
-        self.assertIn("sphinx >= 1.3.1", deps)
-        self.assertIn("twistedchecker >= 0.7.2", deps)
+        self.assertListEqual(
```
Comment (graingert) [inline src/twisted/python/test/test_setup.py]: wouldn't this be easier with:

```python
self.assertEqual(pathlib.Path(_setup.__file__).read_text() == """
    ...
""")
```

Comment (adiroiban) [reply src/twisted/python/test/test_setup.py]: I think that the purpose of this test is to check the high level dependenceis.

We should not care/test for `pillow` or `recommonmark`.
The important thing is to have `sphinx` and maybe `sphinx-rtd-theme` as top dependencies and the test should only check for that.

Comment (graingert) [reply src/twisted/python/test/test_setup.py]: I'm going to delete this test when I upgrade to `setup.cfg`

**Review by adiroiban — APPROVED** [review]
Thanks :)

Wait for all the tests to be green and after that please merge using the standard commit message.

**Pull request merged**